wandb/apis/public.py

```diff
@@ -2236,37 +2236,39 @@ def entity(self):
 
     @property
     def state(self):
+        item = self.get_item()
+        if item:
+            return item["state"].lower()
+
+        raise ValueError(
+            f"Could not find QueuedRunItem associated with id: {self.id} on queue id {self.queue_id} at itemId: {self.id}"
+        )
+
+    @normalize_exceptions
+    def get_item(self):
         query = gql(
             """
-            query GetRunQueueItem($projectName: String!, $entityName: String!, $runQueue: String!) {
+            query GetRunQueueItem($projectName: String!, $entityName: String!, $runQueue: String!, $itemId: ID!) {
```

Comment:
Because we support older versions of the server, this will need to be encapsulated in a try catch mechanism, or use server introspection to see which method we should use.

Reply:
ah cool, yeah makes sense